### ISSUE TITLE
The default advertise_addr can not be set to "0.0.0.0"

### ISSUE THREAD

**Issue by liangping** [open-libra/cli#6]
https://github.com/open-libra/cli/blob/4c40c7d16c26e4e165825bb259e85da3ec6b4950/src/commands/config/builder.rs#L61

0.0.0.0 can be set to listen addr, but not advertised_addr.  
probably need find the local_ip on current server.

**Comment by tarcieri** [comment]
Yeah, the main idea right now is to pass the advertised IP using the '-a' option, which is a required argument. What you're seeing there is a placeholder used by the `Default` impl

**Comment by liangping** [comment]
If you agree, I will add some code to get this address by using ifconfig lib for the Default impl

**Comment by tarcieri** [comment]
There are many cases where the external IP won't be visible to the container/instance. I think it's a lot less error prone to have it explicitly specified.

The `0.0.0.0` used as a default is a placeholder. There is presently no way (at least AFAIK) to actually have `0.0.0.0` emitted in the configuration without it being explicitly specified by the user.

**Comment by liangping** [comment]
Yeah, I know it's not possible. So I think the best way to do is that we ask user to provide a public address if he/she don't specify.  For example : panic with "Advertised Address is required".

**Comment by tarcieri** [comment]
It panics now, albeit with a bad error:

> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParsingError(AddrParseError(()))', src/libcore/result.rs:1084:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

The error message could be improved.
